### ISSUE TITLE
InsertData not preserving formatting on existing spreadsheet

### ISSUE THREAD

**Issue by PeterGrigg** [ClosedXML/ClosedXML#2144]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x] Yes
- [ ] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

e.g. 0.102.0

**What is the current behavior?**
when using insertdata with a DataTable, it is not keeping the formatting of the existing spreadsheet, which version 0.101.0, 0.100.3 and 0.94.5 will preserve.

Complete this.

**What is the expected behavior or new feature?**
Expect it to work like earlier versions

Complete this.

**Is this a regression from the previous version?**
Yes.

Regressions get higher priority. Test against the latest build of the previous minor version. For example, if you experience a problem on v0.95.3, check whether it the problem occurred in v0.94.2 too. 

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
  // CODE to fill table comes from ERP system, too complicated
    private void CopyDataToCell(DataTable dt, string sSheetNameToCopyTo, int row, int col, int rowHeight
      , bool deleteEmptyRows, string colWithFormula) //, int iForceNumRowsToCopy = 0)
    {
      IXLWorksheet ws = this._wbWorkbook.Worksheet(sSheetNameToCopyTo);
      IXLCell cell = ws.Cell(row, col);
      cell.InsertData(dt);
   }
```

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[CM_Full_empty.xlsx](https://github.com/ClosedXML/ClosedXML/files/12284684/CM_Full_empty.xlsx)

**Comment by jahav** [comment]
@igitur Have you even done something merging differential styles? The `XLStyleKey`/`XLStyleValue` have all values set and as far as I can tell, there is no way to differentiate between "alignment is General, because its a default value and user hasn't set anything" and "alignment is General, because it is explicitly set". These two situations case completely different behavior when merging two styles into one. 

Currently, I just have two values and no way to determine which should take precedence.

The problem is that when I am determining a style for the cell that is written, I should either
*  take style that has been explictly set at the cell
* (or if no style is explicitly set), use row style, column style and sheet style and combine them to a single style to use through differential styles. E.g. column style has a custom number format and row has a special fille -> combined style has both.

Instead, I just take first existing from row style, colum style and worksheet style  ([source](https://github.com/ClosedXML/ClosedXML/blob/develop/ClosedXML/Excel/Cells/XLCell.cs#L99)).

The problem is how to merge row+column+sheet style.

**Comment by jahav** [comment]
To fix this properly, I need to

# Add `apply*` to style
Currently, ClosedXML doesn't represents the `apply*` (applyAlignment, applyBorder, applyFill, applyFont, applyNumberForm, applyProtection) flags to styles. These flags are part of style in OOXML and determine how are styles applied, but are missing in the code. Save and load of course must work too.

# Improve style unification.

Currently, a style is determined by a priority (first element that has a style wins): cell, row, column, worksheet. That shoudl be changed to 
* cell
* if cell is missing a style (e.g. blank cell), use row, column and worksheet style to caulculate composite style.

# Add style to cells where needed 
I think that works, or at least half works. I remember seeing a code for that, though likely for bad style unification.

Basically when a style is changed, find all cells that are affected and check that 

Example:
* Initial state: Row says font should be Arial. Column doesn't have apply for font. Thus the cross-section cell will have font arial.
* Style for column is changed, it says font `Consolas`. That collides with row (that has natually precedence) and thus there is a conflict. The last one wins (style set by column should have precedence, because it was set last) and the cross-section cell should have font `Consolas`. But that cell doesn't exist, so it is created and have font set to `Consolas`.
